cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -145,6 +145,21 @@ constexpr bool type_id_matches_device_storage_type(type_id id)
     return Id;                                             \
   }                                                        \
   template <>                                              \
+  constexpr inline type_id type_to_id<Type const>()        \
+  {                                                        \
+    return Id;                                             \
+  }                                                        \
+  template <>                                              \
+  constexpr inline type_id type_to_id<Type volatile>()     \
+  {                                                        \
+    return Id;                                             \
+  }                                                        \
+  template <>                                              \
+  constexpr inline type_id type_to_id<Type const volatile>() \
+  {                                                        \
+    return Id;                                             \
+  }                                                        \
+  template <>                                              \
```

Comment:
nit: is it possible to simply this with `std::decay` or `std::remove_cv`?

Reply:
That's an excellent point!  I did think about that but I couldn't figure out how.  If the type were the **return** then I think so!  But the type is the input and this is a macro.  Let me try...

Reply:
I made an alternative solution.